web/src/site-admin/SiteAdminExternalServiceForm.tsx

```diff
@@ -82,7 +86,8 @@ export class SiteAdminExternalServiceForm extends React.Component<Props, {}> {
                     disabled={this.props.loading}
                 >
                     {this.props.loading && <LoadingSpinner className="icon-inline" />}
-                    {this.props.mode === 'edit' ? 'Update external service' : 'Add external service'}
+                    {this.props.submitName ??
+                        (this.props.mode === 'edit' ? 'Update external service' : 'Add external service')}
```

Comment:
Since we're moving away from "external service" language we should do this consistently for the configuration page too. Not sure if this should be slightly different, other options include: Add/update code host configuration, Configure repositories for <name of code host i.e. GitHub>.

```suggestion
                        (this.props.mode === 'edit' ? 'Update code host' : 'Add code host')}
```